core-common/src/main/java/org/glassfish/jersey/Uri.java

```diff
@@ -13,8 +13,9 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  */
+// Portions Copyright [2018] [Payara Foundation and/or its affiliates]
 
-package org.glassfish.jersey.server;
+package org.glassfish.jersey;
```

Comment:
uri would make sense to put to package org.glassfish.jersey.uri

Reply:
Hi @jansupol ,

As MP REST Client requires core-server#org.glassfish.jersey.server.Uri api, so previously I moved this api to core-common#org.glassfish.jersey.Uri, and now to core-common#org.glassfish.jersey.uri.Uri (after feedback).

But this api is publicly available so it will not be compatibly, if application depends on jersey (instead of JAX-RS api) .

So perhaps, either I should : 
1- Move this api to  core-common#org.glassfish.jersey.server.Uri
or 
2. Create another api  in core-client#org.glassfish.jersey.client.Uri (without refactoring the  core-server#org.glassfish.jersey.server.Uri)

Please suggest.

Reply:
Right, did not think of that. I would move it core-common as you did and retain the one server (just inherit from the common) and mark it as deprecated, remove it in 3.0. But, I cannot find where it is actually used in the PR for which it was needed to be moved.

Reply:
Oops, I forgot Uri is an annotation...I am thinking to create a Parameter in common and a new one in a server, inherit from the common, include the server @uri and use the server Parameter on a server, keep Uri at a server, it is not used by the mp client

Reply:
There is really not a nice solution to this. I tried to come up with [something](https://github.com/eclipse-ee4j/jersey/pull/4091), but I am not particularly happy about the solution. I mentioned some alternatives, not sure which would be right.

Reply:
@jGauravGupta Let's split the PR into more parts. I have separated the inserters classes on your behalf into a separate PR #4098.